tests/queries/0_stateless/01280_ttl_where_group_by.sh

```diff
@@ -1,19 +1,35 @@
-drop table if exists ttl_01280_1;
+#!/usr/bin/env bash
 
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. $CURDIR/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --query "drop table if exists ttl_01280_1"
+
+function optimize()
+{
+    for i in {0..20}; do
+        $CLICKHOUSE_CLIENT --query "OPTIMIZE TABLE $1 FINAL SETTINGS optimize_throw_if_noop=1" 2>/dev/null && break
+        sleep 0.3
+    done
+}
+
+$CLICKHOUSE_CLIENT -n --query "
 create table ttl_01280_1 (a Int, b Int, x Int, y Int, d DateTime) engine = MergeTree order by (a, b) ttl d + interval 1 second delete where x % 10 == 0 and y > 5;
 insert into ttl_01280_1 values (1, 1, 0, 4, now() + 10);
 insert into ttl_01280_1 values (1, 1, 10, 6, now());
 insert into ttl_01280_1 values (1, 2, 3, 7, now());
 insert into ttl_01280_1 values (1, 3, 0, 5, now());
 insert into ttl_01280_1 values (2, 1, 20, 1, now());
 insert into ttl_01280_1 values (2, 1, 0, 1, now());
-insert into ttl_01280_1 values (3, 1, 0, 8, now());
-select sleep(1.1) format Null;
-optimize table ttl_01280_1 final;
-select a, b, x, y from ttl_01280_1;
+insert into ttl_01280_1 values (3, 1, 0, 8, now());"
 
-drop table if exists ttl_01280_2;
+sleep 2
```

Comment:
sleep won't help.

Reply:
Something like a loop with retries and optimize_throw_if_noop will help.

Reply:
Also help me with review of https://github.com/ClickHouse/ClickHouse/pull/11346
We discussed it on today meeting.

Reply:
Sleep is needed because of 1 second TTL. Optimize in a loop also called in the next line.

Reply:
Ok, thank you!